src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -1296,11 +1292,24 @@ public function getReturnRedirectUrlForOrder( WC_Order $order ) {
 
 			}
 
-			$payment = Mollie_WC_Plugin::getPaymentObject()->getActiveMolliePayment($order_id, false );
+
+            try {
+                $payment = $this->activePaymentObject($order_id, false);
+            } catch (UnexpectedValueException $exc) {
+			    $message = 'There was a problem when processing your payment. Please try again.';
```

Comment:
If this is a message to show to the customer in frontend I suggest to make it translatable.